src/include/transaction/transaction_context.h

```diff
@@ -163,7 +163,12 @@ class TransactionContext {
    * @param a the action to be executed. A handle to the system's deferred action manager is supplied
    * to enable further deferral of actions
    */
-  void RegisterAbortAction(const TransactionEndAction &a) { abort_actions_.push_front(a); }
+  void RegisterAbortAction(const TransactionEndAction &a) {
+    // TODO(wuwenw): transaction context is not thread safe for now, and a latch is used here to protect it, may need
```

Comment:
This seems quite hacky and is likely to affect our performance numbers (now, in addition to registering an unnecessary abort action in the "happy path", we take a latch).

Reply:
Was pointed to the slack conversation, I guess this is fine for now.